proxy/proxy.go

```diff
@@ -481,7 +481,7 @@ func mapRequest(ctx *context, requestContext stdlibcontext.Context, removeHopHea
 		setRequestURLFromRequest(u, r)
 		setRequestURLForDynamicBackend(u, stateBag)
 	case eskip.LBBackend:
-		endpoint = setRequestURLForLoadBalancedBackend(u, rt, routing.NewLBContext(r, rt))
+		endpoint = setRequestURLForLoadBalancedBackend(u, rt, &routing.LBContext{Request: r, Route: rt, Params: stateBag})
```

Comment:
Here statebag is used as a parameter map for the `LBContext`. Its a kind of loose coupling between filters and algorithm, is that fine?

Other options I've considered did not look ok:
* stdlib request context needs request substitution in the filter context
* evaluating template within the algorithm requires it to be aware of path variables required by `eskip.TemplateContext`. It also requires route postprocessor

Reply:
For the stdlib case you would use context.WithValue() and you would overwrite somewhere ctx.Request().Context(), right?
So you don't need to overwrite Request, only the context or do I miss something?

For me the current approach seems also fine.

Reply:
https://golang.org/pkg/net/http/#Request.WithContext creates request copy that would need to replace request in filter context

Reply:
this looks fine to me